### ISSUE TITLE
Publish freesound-python in PyPi

### ISSUE THREAD

**Issue by ffont** [MTG/freesound-python#5]
It would be convenient to publish freesound-python in PyPi so it can be easily installed with easy_install or pip. See the instructions here: http://peterdowns.com/posts/first-time-with-pypi.html.

We could wait until we close the other issues, make a 1.0 release of the client (1.0 tag) and then host it.

**Comment by turpaultn** [comment]
Hi there, 
Is it still in the pipe to do a PyPi package ? 

It would be very useful to me because we're using freesound files to generate complex soundscapes for sound event recognition. For now we have a list of freesound files manually created, but we'd like people to be able to do our annotation and curation work to promote new interesting data created.
Having an easy install to add it automatically in the PyPi install of my package: https://github.com/turpaultn/DESED/

I could copy your freesound file, but it seems better in the long run and to manage issues to use your package :)
Let me know your future directions so I can decide accordingly. :)

PS: writing an issue instead of a mail in case other people are in my situation.
Thanks in advance, 
Nicolas

**Comment by ffont** [comment]
Hi Nicolas,

We should spend some time with this, fix a couple of bugs and publish it, but we never seem to find the time. 
However, can't you specify `git+https://github.com/MTG/freesound-python` as a dependency? For example, `pip install git+https://github.com/MTG/freesound-python` works.

**Comment by turpaultn** [comment]
I can understand, wouldn't it be possible to release the current version as a 0.1 for example ? 
Yes, of course,  I'll do it like this for now. Just from experience, in this case I'll need to follow this repo in case some changes are impacting my code (if the version is not updated when changes are made).
Thank you for the answer :)

**Comment by ilesinge** [comment]
I have made a fork of this repository and published it on pypi:
https://pypi.org/project/freesound-api/
The fork lies here: https://github.com/ilesinge/freesound-api
There is no functional change, only packaging and README changes.
If you are interested, I could do a PR on your repo.

**Comment by ffont** [comment]
Hi @ilesinge, that sounds like a good idea, maybe it is time for us to publish it there. Did you change the name so it would not overlap with the official name in case we published it or for some other reason? We would prefer to keep the name in pypi. Thanks!

**Comment by ilesinge** [comment]
Yeah exactly, I changed the name in order to let you use the original name. I'll do a PR.

**Comment by ilesinge** [comment]
See https://github.com/MTG/freesound-python/pull/33 for the suggested implementation with https://python-poetry.org/
May need review:
- pip package name: https://github.com/MTG/freesound-python/pull/33/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R2
- maintainers list: https://github.com/MTG/freesound-python/pull/33/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R6  
- testing the original "git clone & python setup.py install" install method still works

**Comment by ilesinge** [comment]
Oh and in case you never published on PyPi with poetry:
1. Create an account on PyPi if needed
2. Create a PyPI API token: https://pypi.org/manage/account/token/
2. Install Poetry https://python-poetry.org/docs/#installation
3. Configure PyPI to use the token: `poetry config pypi-token.pypi <my-token>`
3. Build the package by running: `poetry build` 
4. Publish the package by running `poetry publish`
And voilà!
